### PR TITLE
Excited groups will care about gas reactions before dismantling. Turfs will not sleep when there are gas reactions. Fixes water vapour not consuming water vapour when freezing turfs.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Excited groups will no longer dismantle when there is a gas reaction occurring, and turfs no longer sleep when there's a gas reaction. Fixes water vapour not consuming water vapour when freezing turfs.

## Why It's Good For The Game
To prevent odd phenomena where turfs with water vapour sleep, then stop applying the slippery effect despite being in conditions to react.

## Changelog

:cl:
fix: Fixes water vapour not consuming water vapour when freezing turfs.
code: Turfs will not sleep during an ongoing gas reaction, and excited groups won't dismantle if one of their turfs are reacting.
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 1]
```diff
@@ -41,6 +41,8 @@
 
 	///If there is an active hotspot on us store a reference to it here
 	var/obj/effect/hotspot/active_hotspot
+	///If there is a reaction, store the result.
+	var/reacting
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:45]: why is this a variable on /turf?

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 2]
```diff
@@ -345,10 +347,12 @@
...
 			enemy_tile.consider_pressure_difference(src, difference)
 
-	our_air.react(src)
+	reacting = our_air.react(src)
+	if(our_excited_group)
+		our_excited_group.has_reactions |= reacting
 
 	update_visuals()
-	if(!consider_superconductivity(starting = TRUE) && !active_hotspot) //Might need to include the return of react() here
+	if(!consider_superconductivity(starting = TRUE) && !active_hotspot && !reacting) //Might need to include the return of react() here
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:355]: remove my old comment, also if you're gonna check the flags in excited group handling do it here too

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 3]
```diff
@@ -407,6 +411,8 @@
 	var/display_id = 0
 	///Wrapping loop of the index colors
 	var/static/wrapping_id = 0
+	///If there are turfs in the group with gasmixtures with reactions occuring.
+	var/has_reactions = NONE
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:415]: call this turf_reactions, and describe it as all turf reaction flags we've received. what you've got now is a bit unclear

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 4]
```diff
@@ -109,8 +109,9 @@
 	switch(air.temperature)
 		if(-INFINITY to WATER_VAPOR_DEPOSITION_POINT)
 			if(location?.freeze_turf())
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:111]: a: does this mean if we can't freeze the turf we won't react? What does that mean?

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 5]
```diff
@@ -109,8 +109,9 @@
...
-				SET_REACTION_RESULTS(0)
-			. = REACTING
+				air.gases[/datum/gas/water_vapor][MOLES] -= MOLES_GAS_VISIBLE
+				SET_REACTION_RESULTS(MOLES_GAS_VISIBLE)
+				. = REACTING
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:114]: B: this is duped from below right? why not pull it out?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 4 above)
Comment (Pickle-Coding) [reply code/modules/atmospherics/gasmixtures/reactions.dm:111]: The reaction isn't really doing anything if that check doesn't pass, so I wouldn't really consider that as reacting.

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 6]
```diff
@@ -345,10 +347,12 @@
 		else
 			enemy_tile.consider_pressure_difference(src, difference)
 
-	our_air.react(src)
+	reacting = our_air.react(src)
+	if(our_excited_group)
+		our_excited_group.has_reactions |= reacting
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:352]: make a comment describing why we do this, somewhere

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 4 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:112]: oh TRUE then we don't set the flag
yeah nevermind this is safe

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 7]
```diff
@@ -345,10 +345,12 @@
...
 			enemy_tile.consider_pressure_difference(src, difference)
 
-	our_air.react(src)
+	var/reacting = our_air.react(src)
+	if(our_excited_group)
+		our_excited_group.turf_reactions |= reacting //Adds the flag to turf_reactions so excited groups can check for them before dismantling.
 
 	update_visuals()
-	if(!consider_superconductivity(starting = TRUE) && !active_hotspot) //Might need to include the return of react() here
+	if(!consider_superconductivity(starting = TRUE) && !active_hotspot && !(reacting & (REACTING | STOP_REACTIONS)))
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:353]: if you wanna be cool make a define out of  REACTING | STOP_REACTIONS that explains more of what it actually represents

**Pull request merged**